### ISSUE TITLE
How to pass babel properties to the babel-eslint parser?

### ISSUE THREAD

**Issue by gajus** [babel/babel-eslint#198]
It appears that babel-eslint parser is looking for ".babelrc" in the current working directory.

How do I make babel-eslint parser ignore the ".babelrc" in the current working directory and pass custom setting, e.g. "stage" 0?

**Comment by gajus** [comment]
The use case is https://github.com/gajus/pragmatist tool. `pragmatist` abstracts common build operations (such as "babel" tranpilation). As a result, the main project does not have ".babelrc" file in the main directory.

**Comment by hzoo** [comment]
We don't have a way to do that currently - but this is the same request as https://github.com/babel/babel-eslint/issues/171 right?

**Comment by gajus** [comment]
Maybe. I read that thread but could not figure out what it is about.

**Comment by gajus** [comment]
I am reading the source code, but I cannot figure out what would I need to patch to allow custom options. I am happy to provide a PR with a little help.

**Comment by gajus** [comment]
This is the part where you are using babel to parse the source, https://github.com/babel/babel-eslint/blob/master/index.js#L394. Which refers to the https://github.com/babel/babel/blob/master/packages/babel/src/api/node.js#L81-L106. But I cannot see where the https://babeljs.io/docs/usage/options/ are being parsed.

**Comment by gajus** [comment]
I am assuming these are not the same because the latter source code includes not documented properties, e.g.

```js
opts.allowHashBang = true;
opts.sourceType = "module";
opts.ecmaVersion = Infinity;
```

**Comment by hzoo** [comment]
Well you use babel-eslint because eslint allows `parser: esprima`. You just need to export a parse function that allows passing in the code which is what https://github.com/babel/babel-eslint/blob/master/index.js#L372 is. However, I think don't have a way to parse any of the options given to babel-eslint unless we do something like read from a local `.bablelrc` or something.

**Comment by gajus** [comment]
There is no official way to do it (https://github.com/eslint/eslint/issues/4264#issuecomment-151254760). What about using global variables? It seems like one of those rare cases where a global variable would be a suitable workaround until ESLint CLIEngine supports passing configuration being passed to the parser, e.g.

```js
global.BabelESLintParser = global.BabelESLintParser || {};
global.BabelESLintParser.BabelConfig = {
    stage: 0
};
```

**Comment by gajus** [comment]
Or if there was a way to set which `.babelrc` file Babel should be using. Either would work.

**Comment by hzoo** [comment]
Goina close since this is similar to https://github.com/babel/babel-eslint/issues/171
